pandas/tests/tools/test_to_datetime.py

```diff
@@ -2923,17 +2941,22 @@ def test_incorrect_value_exception(self):
                 to_datetime(["today", "yesterday"])
 
     @pytest.mark.parametrize(
-        "format, warning", [(None, UserWarning), ("%Y-%m-%d %H:%M:%S", None)]
+        "format, warning",
+        [
+            (None, UserWarning),
+            ("%Y-%m-%d %H:%M:%S", None),
+            ("%Y-%d-%m %H:%M:%S", None),
+        ],
     )
     def test_to_datetime_out_of_bounds_with_format_arg(self, format, warning):
         # see gh-23830
         msg = (
-            "Out of bounds nanosecond timestamp: 2417-10-27 00:00:00 "
-            "present at position 0"
+            r"^Out of bounds nanosecond timestamp: 2417-10-10 00:00:00 "
+            r"present at position 0$"
         )
         with pytest.raises(OutOfBoundsDatetime, match=msg):
             with tm.assert_produces_warning(warning, match="Could not infer format"):
-                to_datetime("2417-10-27 00:00:00", format=format)
+                to_datetime("2417-10-10 00:00:00", format=format)
```

Comment:
slightly changing the input here so we can parametrise of it with both `"%Y-%m-%d %H:%M:%S"` and `"%Y-%d-%m %H:%M:%S"` (to check ISO vs non-ISO)